checker/src/main/java/org/checkerframework/checker/mustcall/MustCallTransfer.java

```diff
@@ -0,0 +1,351 @@
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitTernaryExpression(
+      TernaryExpressionNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitTernaryExpression(node, input);
+    updateStoreWithTempVar(result, node);
+    return result;
+  }
+
+  /**
+   * Adds newAnno as the value for target to all stores contained in result.
+   *
+   * @param result a TransferResult containing one or more stores
+   * @param target a JavaExpression whose type is being modified
+   * @param newAnno the new type for target
+   */
+  public void insertIntoStores(
+      TransferResult<CFValue, CFStore> result, JavaExpression target, AnnotationMirror newAnno) {
+    if (result.containsTwoStores()) {
+      CFStore thenStore = result.getThenStore();
+      CFStore elseStore = result.getElseStore();
+      thenStore.insertValue(target, newAnno);
+      elseStore.insertValue(target, newAnno);
+    } else {
+      CFStore store = result.getRegularStore();
+      store.insertValue(target, newAnno);
+    }
+  }
+
+  /**
+   * Returns the arguments of the @CreatesObligation annotation on the invoked method, as
+   * JavaExpressions. Returns the empty set if the given method has no @CreatesObligation
+   * annotation.
+   *
+   * <p>If any expression is unparseable, this method reports an error and returns the empty set.
+   *
+   * @param n a method invocation
+   * @param atypeFactory the type factory to report errors and parse the expression string
+   *     recomputation
+   * @param supplier a type factory that can supply the executable elements for CreatesObligation
+   *     and CreatesObligation.List's value elements. Usually, you should just pass atypeFactory
+   *     again. The arguments are different so that the given type factory's adherence to both
+   *     protocols are checked by the type system.
+   * @return the arguments of the method's @CreatesObligation annotation, or the empty set
+   */
+  public static Set<JavaExpression> getCreatesObligationExpressions(
+      MethodInvocationNode n,
+      GenericAnnotatedTypeFactory<?, ?, ?, ?> atypeFactory,
+      CreatesObligationElementSupplier supplier) {
+    AnnotationMirror createsObligationList =
+        atypeFactory.getDeclAnnotation(n.getTarget().getMethod(), CreatesObligation.List.class);
+    if (createsObligationList != null) {
+      // Handle a set of create obligation annotations.
+      List<AnnotationMirror> createsObligations =
+          AnnotationUtils.getElementValueArray(
+              createsObligationList,
+              supplier.getCreatesObligationListValueElement(),
+              AnnotationMirror.class);
+      Set<JavaExpression> results = new HashSet<>();
```

Comment:
Using a HashSet is expensive and gratuitous.  I would create `new ArrayList<>(1)`, and call `contains` before `add`.  This will save both space and time, because the set of targets is usually small.